src/main/java/ca/corefacility/bioinformatics/irida/ria/web/services/UIPasswordResetService.java

```diff
@@ -0,0 +1,86 @@
+package ca.corefacility.bioinformatics.irida.ria.web.services;
+
+import java.security.Principal;
+import java.util.Locale;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.MessageSource;
+import org.springframework.mail.MailSendException;
+import org.springframework.stereotype.Component;
+import org.springframework.web.bind.annotation.PathVariable;
+
+import ca.corefacility.bioinformatics.irida.model.user.PasswordReset;
+import ca.corefacility.bioinformatics.irida.model.user.User;
+import ca.corefacility.bioinformatics.irida.ria.web.PasswordResetController;
+import ca.corefacility.bioinformatics.irida.service.EmailController;
+import ca.corefacility.bioinformatics.irida.service.user.PasswordResetService;
+import ca.corefacility.bioinformatics.irida.service.user.UserService;
+
+/**
+ * Handles service calls for password resets.
+ */
+@Component
+public class UIPasswordResetService {
+	private static final Logger logger = LoggerFactory.getLogger(UIPasswordResetService.class);
+	private final UserService userService;
+	private final PasswordResetService passwordResetService;
+	private final EmailController emailController;
+	private final MessageSource messageSource;
+
+	@Autowired
+	public UIPasswordResetService(UserService userService, PasswordResetService passwordResetService,
+			EmailController emailController, MessageSource messageSource) {
+		this.userService = userService;
+		this.passwordResetService = passwordResetService;
+		this.emailController = emailController;
+		this.messageSource = messageSource;
+	}
+
+	/**
+	 * Create a new {@link PasswordReset} for the given {@link User}
+	 *
+	 * @param userId    The ID of the {@link User}
+	 * @param principal a reference to the logged in user.
+	 * @param locale    a reference to the locale specified by the browser.
+	 * @return text to display to the user about the result of creating a password reset.
+	 * @throws Exception if there is an error emailing the password reset.
+	 */
+	public String adminNewPasswordReset(@PathVariable Long userId, Principal principal, Locale locale)
```

Comment:
The controller already gets the `userId` from the PathVariable and then passes it to this method. So you can remove the `@PathVariable` here infront of `userId`

Reply:
I dunno how that happened. Please see 8abd09e5092fc9e88fabdbc6ac937de570ace856.